google-cloud-spanner/acceptance/spanner/client/transaction_test.rb

```diff
@@ -126,6 +128,83 @@
       assert_accounts_equal expected, actual
     end
   end
+  focus
+  it "supports tx isolation with read and update" do
+    results = db.read "accounts", [:reputation], keys: 1, limit: 1
+    original_val = results.rows.first[:reputation]
+    begin
+      latch = Concurrent::CountDownLatch.new 1
+      thr_1 = Thread.new do
+        db.transaction do |tx|
+          tx_results = tx.read "accounts", [:reputation], keys: 1, limit: 1
+          puts "read 1"
+          tx_val = tx_results.rows.first[:reputation]
+          new_val = tx_val + 1
+          tx.update "accounts", [{ account_id: 1, reputation: new_val }]
+          puts "write 1"
+          latch.wait
```

Comment:
>  @blowmage is correct in that you should wait before the commit. 

The current position of the wait is just before the end of the block, and therefore just before the commit. @blowmage asked:

> Don't you want to wait before you do the update?